common/ffsparser.cpp

```diff
@@ -5106,7 +5089,7 @@ USTATUS FfsParser::parseCpdExtensionsArea(const UModelIndex & index)
     UINT32 offset = 0;
     while (offset < (UINT32)body.size()) {
         const CPD_EXTENTION_HEADER* extHeader = (const CPD_EXTENTION_HEADER*) (body.constData() + offset);
-        if (extHeader->Length <= ((UINT32)body.size() - offset)) {
```

Comment:
This is what fixes the hang, but it's still not perfect. All this shit had been written in hurry for a specific case and begs for a complete rewrite.

Reply:
Yes… but should do for the time being.